sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/KinesisPartitioner.java

```diff
@@ -47,6 +47,26 @@
     return null;
   }
 
+  /**
+   * An explicit partitioner that always returns a {@code Nonnull} explicit hash key. The partition
+   * key is irrelevant in this case, though it cannot be {@code null}.
+   */
+  interface ExplicitPartitioner<T> extends KinesisPartitioner<T> {
+    @Override
+    default @Nonnull String getPartitionKey(T record) {
+      return "a"; // will be ignored, but can't be null
```

Comment:
Return just empty string?

Reply:
It can't be empty, this constant is apparently what KPL uses in this case

Reply:
Then it would be better to extract as a constant with a proper javadoc.

Reply:
This value is really totally irrelevant and has no meaning (as long as it doesn't fail validation checks), imho extracting it into a constant just overloads it with meaning it doesn't have...